internal/abci/abci.go

```diff
@@ -283,16 +339,23 @@ func (a *AbciApp) FinalizeBlock(ctx context.Context, req *abciTypes.RequestFinal
 		res.TxResults = append(res.TxResults, abciRes)
 	}
 
-	res.ConsensusParamUpdates = &tendermintTypes.ConsensusParams{ // why are we "updating" these on every block? Should be nil for no update.
-		// we can include evidence in here for malicious actors, but this is not important this release
-		Version: &tendermintTypes.VersionParams{
-			App: a.cfg.ApplicationVersion, // how would we change the application version?
-		},
-		Validator: &tendermintTypes.ValidatorParams{
-			PubKeyTypes: []string{"ed25519"},
-		},
+	// If at activation height, submit any consensus params updates associated
+	// with the fork. They should not overlap (some forks should be superseded
+	// by later fork definitions and not used on new networks).
+	paramUpdates := defaultParamUpdates()
```

Comment:
In `defaultParams`, you specify ed25519 as the only key type. Wouldn't you just want a totally empty params here, so that any specially configured values for the `PubKeyTypes` won't be overwritten by this default value?

Reply:
Oh, so we update cometbft at every block? Damn that seems really messy

Reply:
Since we're basically starting fresh with 0.8, I'm fixing this longstanding quirk.  Using totally empty params as the updates, as is intended when nothing changes.